component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -83,13 +83,14 @@ private RecordImpl(final Map<String, Object> values, final Schema schema) {
 
     @Override
     public <T> T get(final Class<T> expectedType, final String name) {
-        final Object value = values.get(name);
+        final String entryName = Schema.sanitizeConnectionName(name);
+        final Object value = values.get(entryName);
```

Comment:
I don't think that it's wise to apply a sanitzation again for getting values.
We cannot ensure that the result of ` Schema.sanitizeConnectionName(name)` will return the same sanitize made in builder...

What if I change a bit one of your test below like that?
```
       final Entry entry1 = new EntryImpl.BuilderImpl() //
                .withName("1 name a") //
                .withType(Type.DATETIME) //
                .withDefaultValue(null) //
                .build();
        final Entry entry2 = new EntryImpl.BuilderImpl() //
                .withName("2 name a") //
                .withType(Type.DOUBLE) //
                .withDefaultValue(11.11) //
                .build();
...
```
If 
- you call ` Schema.sanitizeConnectionName("1 name a")`  you'll get `_name_a`.
- you call ` Schema.sanitizeConnectionName("2 name a")`  you'll get `_name_a`.